src/sagas.js

```diff
@@ -241,6 +253,38 @@ function* fetchAllVms (action) {
   yield put(persistState())
 }
 
+function* fetchAllPools (action) {
+  const allPools = yield callExternalAction('getAllPools', Api.getAllPools, action)
+
+  if (allPools && allPools['vm_pool']) { // array
+    const internalPools = allPools.vm_pool.map(pool => Api.poolToInternal({ pool }))
+
+    const poolIdsToPreserve = internalPools.map(pool => pool.id)
+    yield put(removeMissingPools({ poolIdsToPreserve }))
+
+    yield put(updatePools({ pools: internalPools }))
+    yield put(updateVmsPoolsCount())
```

Comment:
Better if it will be separately, cause of more flexibility, and less repeating in functions that can use updateVm and removeVm together or separately (depends on conditions)